src/analysis/analyzer.ts

```diff
@@ -25,21 +25,27 @@ export class Analyzer extends AnalyzerGen implements vs.Disposable {
 		let args = [];
 
 		// Optionally start Observatory for the analyzer.
-		let observatoryPort = config.analyzerObservatoryPort;
-		if (observatoryPort)
-			args.push(`--observe=${observatoryPort}`);
+		if (config.analyzerObservatoryPort)
+			args.push(`--observe=${config.analyzerObservatoryPort}`);
 
 		args.push(analyzerPath);
 
 		// Optionally start the analyzer's diagnostic web server on the given port.
-		let diagnosticsPort = config.analyzerDiagnosticsPort;
-		if (diagnosticsPort)
-			args.push(`--port=${diagnosticsPort}`);
+		if (config.analyzerDiagnosticsPort)
+			args.push(`--port=${config.analyzerDiagnosticsPort}`);
 
 		// Add info about the extension that will be collected for crash reports etc.
 		args.push(`--client-id=DanTup.dart-code`);
 		args.push(`--client-version=${extensionVersion}`);
 
+		// The analysis server supports a verbose instrumentation log file.
+		if (config.analyzerInstrumentationLogFile)
+			args.push(`--instrumentation-log-file=${config.analyzerInstrumentationLogFile}`);
```

Comment:
The path of the VM gets passed to `child_process.spawn` as its own arg, so it didn't need it (I think we tested it when #106 was done). I don't think any of the other args here have been strings before (at least, not if you enter them correctly) so it hasn't come up.

I can do some testing of this when I get some time at a PC (I really ought to give Linux+Code a go on my Chromebook, would get many more opportunities to spend time on this!); will leave this open as a reminder, but otherwise it looks fine to me!